### PR TITLE
kubernetes_use_configmap

### PR DESCRIPTION
**Pull request by ReSearchITEng**
fixes https://github.com/zalando/postgres-operator/issues/855 bug, which causes missing selectors in master service, when they are actually required.
This way, this PR aligns operator with Patroni's DSC method: k8s with configMaps, by solving the bug in this "if condition": https://github.com/zalando/postgres-operator/blob/07c5da35e3572b8302f33e3dd8fbf5e8ca846229/pkg/cluster/k8sres.go#L1339
 
Because the operator was mainly tested in k8s with the default "endpoints" option, this use case (configMaps) option was hidden till tests were done in OpenShift.
When the kubernetes_use_configmaps parameter is set to true, this PR will also export the required **patroni** environment variable with the same name:  KUBERNETES_USE_CONFIGMAPS=true (as per patroni docs).

### CODE REVIEW

File: pkg/util/config/config.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ type Config struct {
 	ConnectionPool
 
 	WatchedNamespace      string            `name:"watched_namespace"`    // special values: "*" means 'watch all namespaces', the empty string "" means 'watch a namespace where operator is deployed to'
+	KubernetesUseConfigMaps                bool              `name:"kubernetes_use_configmaps" default:"false"`
```
Comment (FxKu) [inline pkg/util/config/config.go]: please `go.fmt` all your code. Usually, your IDE should auto-align here.

File: charts/postgres-operator/values.yaml [hunk 2]
```diff
@@ -23,6 +23,8 @@ configGeneral:
   enable_shm_volume: "true"
   # etcd connection string for Patroni. Empty uses K8s-native DCS.
   etcd_host: ""
+  # Select if setup uses endpoints (default), or configmaps to manage leader (DCS=k8s)
+  # kubernetes_use_configmaps: false
```
Comment (FxKu) [inline charts/postgres-operator/values.yaml]:
```suggestion
  # kubernetes_use_configmaps: "false"
```
For configmap we have to quote the bools.

File: manifests/configmap.yaml [hunk 3]
```diff
@@ -43,6 +43,7 @@ data:
   # enable_team_superuser: "false"
   enable_teams_api: "false"
   # etcd_host: ""
+  # kubernetes_use_configmaps: false
```
Comment (FxKu) [inline manifests/configmap.yaml]:
```suggestion
  # kubernetes_use_configmaps: "false"
```

File: manifests/postgresql-operator-default-configuration.yaml [hunk 4]
```diff
@@ -5,6 +5,7 @@ metadata:
 configuration:
   # enable_crd_validation: true
   etcd_host: ""
+  # kubernetes_use_configmaps: "False"
```
Comment (FxKu) [inline manifests/postgresql-operator-default-configuration.yaml]:
```suggestion
  # kubernetes_use_configmaps: false
```
here it's the opposite :smiley:

File: pkg/util/config/config.go (hunk 1 above)
Comment (ReSearchITEng) [reply pkg/util/config/config.go]: oh, right, not sure why it didn't.
Did it now for all of them.

**Comment by Jan-M** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Pull request merged**